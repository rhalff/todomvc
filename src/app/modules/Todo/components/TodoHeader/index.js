// @flow
import React, { PureComponent } from 'react'
import styles from './scss/TodoHeader.scss'
import InputField from '../InputField'
import ToggleComplete from '../ToggleComplete'
import { allComplete } from '../../lib'
import type { Todos } from '../../types/todos'

type Props = {
  items: Todos,
  onCreate: Function,
  onToggleComplete: Function,
  placeholder: string
}

class TodoHeader extends PureComponent<Props> {
  toggleComplete = ({ target }: SyntheticEvent<HTMLInputElement>) => {
    if (target instanceof HTMLInputElement) {
      this.props.onToggleComplete(target.checked)
    }
  }

  render() {
    const { items, onCreate, placeholder } = this.props

    return (
      <header className={styles.todoHeader}>
        {items.length > 0 && (
          <ToggleComplete
            onChange={this.toggleComplete}
            checked={allComplete(items)}
          />
        )}
        <InputField placeholder={placeholder} onSubmit={onCreate} />
        <div />
      </header>
    )
  }
}

export default TodoHeader
