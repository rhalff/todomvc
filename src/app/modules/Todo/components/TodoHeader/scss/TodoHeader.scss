.todoHeader {
  display: flex;
  justify-content: space-between;
  align-items: center;

  > {
    :first-child {
      flex: 0 0 auto;
    }

    :last-child {
      position: absolute;
      right: 0;
    }

    :nth-child(2) {
      flex: 1 1 auto;
    }
  }
}
