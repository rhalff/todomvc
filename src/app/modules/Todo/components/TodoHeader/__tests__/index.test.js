// @flow
import React from 'react'
import TodoHeader from '../'
import renderer from 'react-test-renderer'

const todos = [{ id: 1, text: 'Test Todo' }, { id: 2, text: 'Test Todo 2' }]

test('Can show Todo Header', () => {
  const component = renderer.create(
    <TodoHeader
      items={todos}
      onCreate={() => {}}
      placeholder="What needs to be done?"
    />
  )

  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
