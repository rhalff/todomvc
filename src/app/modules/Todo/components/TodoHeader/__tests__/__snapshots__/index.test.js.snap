// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Can show Todo Header 1`] = `
<header
  className="todoHeader"
>
  <input
    checked={false}
    className="unchecked"
    onChange={[Function]}
    type="checkbox"
  />
  <form
    onSubmit={[Function]}
  >
    <input
      className="edit"
      onChange={[Function]}
      placeholder="What needs to be done?"
      type="text"
      value=""
    />
  </form>
  <div />
</header>
`;
